js/autotrack/reactNavigation.js

```diff
@@ -78,14 +110,33 @@ export const withReactNavigationAutotrack = track => AppContainer => {
                 'Heap: React Navigation is instrumented for autocapture.'
               );
               this.topLevelNavigator = navigatorRef;
-              this.trackInitialRoute();
+              if (this.topLevelNavigator.state) {
+                // We're on React Navigation 4, so track the initial route now.
+                this.trackInitialRouteForState(
+                  this.topLevelNavigator.state.nav
+                );
+              }
             }
           })}
+          onReady={(...args) => {
+            this.captureOnReady();
+
+            if (typeof onReady === 'function') {
```

Comment:
are onReady and onStateChange built-in RN functions? Not familiar with those methods

Reply:
These are input props for the `NavigationContainer` that the HOC is wrapping (see [`onReady`](https://reactnavigation.org/docs/navigation-container#onready) and [`onStateChange`](https://reactnavigation.org/docs/navigation-container#onstatechange))